sdk/textanalytics/Azure.AI.TextAnalytics/src/HealthcareEntity.cs

```diff
@@ -1,20 +1,144 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
 using System.Collections.Generic;
-using Azure.Core;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Azure.AI.TextAnalytics.Models;
 
 namespace Azure.AI.TextAnalytics
 {
     /// <summary>
-    /// HealthcareEntity.
+    /// .
     /// </summary>
-    [CodeGenModel("HealthcareEntity")]
-    public partial class HealthcareEntity
+    public class HealthcareEntity
     {
+        internal HealthcareEntity(HealthcareEntityInternal entity, IEnumerable<HealthcareEntityInternal> healthcareEntities = default, IEnumerable<HealthcareRelationInternal> healthcareRelations = default)
+        {
+            Category = entity.Category;
+            Text = entity.Text;
+            SubCategory = entity.Subcategory;
+            ConfidenceScore = entity.ConfidenceScore;
+            Offset = entity.Offset;
+            DataSources = entity.Links;
+            if (healthcareEntities != null && healthcareRelations != null)
+            {
+                RelatedEntities = ResolveRelatedEntities(entity, healthcareEntities, healthcareRelations);
+            }
+        }
+
+        private static IReadOnlyDictionary<HealthcareEntity, HealthcareEntityRelationType> ResolveRelatedEntities(HealthcareEntityInternal entity,
```

Comment:
it is ok to return just a Dictionary here. We only use the interfaces for properties